src/components/App.js

```diff
@@ -1,14 +1,42 @@
-import React from 'react';
+import React, { Component } from 'react';
 import './App.css';
 import Display from './Display';
 import ButtonPanel from './ButtonPanel';
 import calculate from '../logic/calculate';
 
-const App = () => (
-  <>
-    <Display result={calculate([0, 0, '+'], '+').total} />
-    <ButtonPanel />
-  </>
-);
+class App extends Component {
+  constructor(props) {
+    super(props);
+    this.props = props;
+    this.state = {
+      total: null,
+      next: null,
+      operation: null,
+    };
+  }
+
+  handleClick(buttonName) {
+    const { total, next, operation } = calculate(this.state, buttonName);
+    this.setState({
+      total,
+      next,
+      operation,
+    });
+  }
```

Comment:
- [ ] move `handleClick` inside the `constructor`

Reply:
Done. Although I think it is not a good practice. It would be better to add `handleClick()` method as a class method and binding ( with `.bind(this)` ) it inside the constructor.